extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -451,8 +451,17 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			let parentBookPath: string = notebookFolders.slice(0, notebookFolders.length - depthOfNotebookInBook).join('/');
 			let bookItemToExpand = parentBook.bookItems.find(b => b.tooltip.indexOf(parentBookPath) > -1) ??
 				parentBook.bookItems.find(b => path.relative(notebookPath, b.tooltip)?.split(path.sep)?.length === depthOfNotebookInBook);
+
 			if (!bookItemToExpand) {
-				break;
+				// if the book isn't found, check if the book is in the same level as the parent
+				// since bookItems will not have them yet, check the sections->file property
+				bookItemToExpand = parentBook.bookItems.find(b => b.sections?.find(n => notebookFolders[notebookFolders.length - depthOfNotebookInBook - 1].indexOf(n.file.substring(n.file.lastIndexOf('/') + 1)) > -1));
+				// book isn't found even in the same level, break out and return.
```

Comment:
This is one scenario - but this still isn't enough to fix this completely. What if I have the book item nested under a different child item? 

And then there's the issue in that we are only walking the tree to the "depth" of the Notebook, but that's the depth of the relative path which potentially has nothing to do with the depth in the actual book structure. 

This all just goes back to the book structure having pretty much nothing to do with actual file structure and so making any assumptions of that isn't the way we want to be approaching this.

Reply:
I agree that this would fail in a scenario where the parent node is nested inside of its child node, but that scenario might happen lest than 1% of the time. It's almost always in the same level or above the child nodes. That being said, Livesite has pretty weird wiring of books so it can happen. I agree that we should refactor this to walk through the logical book structure. I wanted to get this in before I can get to it since it's been a critical ask and wouldn't want them to complain it fails again.
And regarding the logical book structure, we currently don't have it and I would like to sync with you on implementation. 
@chlafreniere  Does that sound okay with you?